### DISCUSSION TITLE
S1D0 - Kickoff und Einrichtung der Entwicklungsumgebung

### DISCUSSION THREAD

**Discussion by stan1025** [VDI-CodING/codeING-main#18] [General]
Willkommen zum VDI CodING - Kickoff!

Wir, das Team der VDI CodING Dojos, begrüßt euch hier im Workspace!

Bei Fragen während des Kickoffs bitte hier fragen, damit die Fragen und Antworten entsprechend dokumentiert werden.

**Comment by Blumi102** [comment]
Hallo Andreas :)

**Comment by su-ro** [comment]
Hey zusammen,
ich hab den Thread hier leider zu spät gesehen, sorry für den Doppelpost.

Ich komme bei der Installation von typescript nicht weiter.

```
package name: (hackathon2021) npm install -D typescript
Sorry, name can only contain URL-friendly characters and name can no longer contain capital letters.
```

Habt ihr Ideen, wie ich weiterkomme?

Liebe Grüße
Susi

**Reply by Zangarus** [reply]
Du bist noch beim Ausführen von `npm init`. 

Hier kannst du einfach Enter drücken, um die Standardeinstellungen zu übernehmen. Erst wenn `npm init` beendet wurde, ist der Befehel `npm install -D typescript` auszuführen.

**Comment by olemir** [comment]
Wie hängen die Komponenten zusammen? Also: eslint, packagemanager, npm, node.js

**Reply by Zangarus** [reply]
`npm` ist der Packagemanager den wir nutzen. Die bedeutet, dass wir uns nicht selber um die Installation kümmern, sondern nur npm sagen müssen, was wir gerne für Dependencies nutzen möchten.

`node.js` ist einfach nur eine Laufzeitumgbung, die wir nutzen, um Typescript (bzw. letztlich eigentlich JavaScript) auszuführen. Wir können hiermit auf sehr einfach Art einen Webserver bauen.

`eslint` ist ein Analysewerkzeug für unseren Code. Hier können wir einen Style vorgeben in dem wir programmieren möchten (z.B. keine unnötigen Leerzeichen, keine optionalen Semikolons) und das Tool zeigt uns an, ob wir uns daran halten. Unsere IDE Visual Studio Code korrigiert dann sogar falschen Style automatisch wenn man das Auto Formatting aktiviert.

**Comment by stan1025** [comment]
Der Link zur Plattform mit dem Video ist online: [Video-Plattform](https://1drv.ms/u/s!Apib2MdOJ2WvtowzTb6CDyAlCSZ8xw?e=m7mWw9)
